### PR TITLE
Ret 2723 Added check to welsh tranlations that nothing is missing

### PR DESCRIPTION
**Pull request by paul-peasegood**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,4 @@ smoke-output/reports
 src/test/api/pact/pacts/
 src/test/end-to-end/functional-output/*
 /dump.rdb
+/missingTranslationKeys.txt
```
Comment (marcjudge) [inline .gitignore:34]: This can be removed

File: src/test/unit/translations.spec.ts [hunk 2]
```diff
@@ -65,4 +65,27 @@ describe('Check missing keys in translation files', () => {
 
     expect(findMissingKeys(englishContents, welshContents)).toEqual([]);
   });
+
+  function checkWelshTranlationFile(obj: { [index: string]: unknown }, full_key: string) {
+    for (const k in obj) {
+      if (typeof obj[k] === 'string') {
+        const val: string = obj[k] as string;
+        if (val.includes('Welsh Translation required') || val === '') {
```
Comment (marcjudge) [inline src/test/unit/translations.spec.ts:73]: There is another scenario I've discovered which is the text includes '- Requires Update'

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/test/unit/translations.spec.ts (hunk 2 above)
Comment (ikirsanov) [reply src/test/unit/translations.spec.ts:73]: 'Update Required' also presents in jsons

**Review by marcjudge — APPROVED** [review]

File: src/test/unit/translations.spec.ts [hunk 3]
```diff
@@ -65,4 +65,32 @@ describe('Check missing keys in translation files', () => {
...
   });
+
+  function checkWelshTranlationFile(obj: { [index: string]: unknown }, full_key: string) {
+    for (const k in obj) {
+      if (typeof obj[k] === 'string') {
+        const val: string = obj[k] as string;
+        if (
+          val.includes('Welsh Translation required') ||
+          val.includes('Requires Update') ||
+          val.includes('Requires Update') ||
```
Comment (marcjudge) [inline src/test/unit/translations.spec.ts:76]: Same value included twice

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/test/unit/translations.spec.ts (hunk 2 above)
Comment (paul-peasegood) [reply src/test/unit/translations.spec.ts]: done

File: src/test/unit/translations.spec.ts (hunk 3 above)
Comment (paul-peasegood) [reply src/test/unit/translations.spec.ts]: fixed

**Review by ikirsanov — APPROVED** [review]

**Review by marcjudge — APPROVED** [review]

**Comment by marcjudge** [comment]
Seems to a strange issue with communication between Github and Jenkins not marking checks as complete. Build and deployment has completed so I'm bypassing branch protections.

**Pull request merged**